src/mpid/ch4/src/ch4_impl.h

```diff
@@ -624,9 +618,8 @@ static inline int MPIDIU_valid_group_rank(MPIR_Comm * comm, int rank, MPIR_Group
     do {                                                                               \
         MPIDIG_EPOCH_CHECK_SYNC(win, mpi_errno, goto fn_fail);                     \
         MPIDIG_EPOCH_OP_REFENCE(win);                                              \
-        /* Check target sync status for any target_rank except PROC_NULL. */           \
-        if (target_rank != MPI_PROC_NULL)                                              \
-            MPIDIG_EPOCH_CHECK_TARGET_SYNC(win, target_rank, mpi_errno, goto fn_fail);  \
+        /* Check target sync status for any target_rank. */           \
+        MPIDIG_EPOCH_CHECK_TARGET_SYNC(win, target_rank, mpi_errno, goto fn_fail);  \
     } while (0);
```

Comment:
I thought about this, but this change will not be valid alone if we had not done moving the `MPI_PROC_NULL` checks to `MPI_` for RMA routines. So it might be better to stay with the latter changes.
Other than this, I have addressed your other comments. The travis builds are passed (I think we may not need another round of Jenkins test). Could you please take another look?

Reply:
I do not think so. The reason we can move PROC_NULL check to `MPI_` layer is because we decided to give up error checking for PROC_NULL in RMA. So the change in error checking (and RMA code) should go first. Then you move all PROC_NULL checking to `MPI_` layer in the second commit.

Reply:
> The reason we can move PROC_NULL check to `MPI_` layer is because we decided to give up error checking for PROC_NULL in RMA.

I might be missing some point, but the original code here already means no error checking for `PROC_NULL`, right? So we are not "giving up" error checking for `PROC_NULL` here, but just removing an unnecessary check **after** we have done the moving action, so we know that an RMA op with `PROC_NULL` will not reach here. If the moving action had not been done, this change alone is invalid because `MPIDIG_EPOCH_CHECK_TARGET_SYNC` might not be able to handle `PROC_NULL`.

Reply:
In original code, we still check error in MPIDIG_EPOCH_CHECK_SYNC for PROC_NULL, but just do not check per-target status. Now, you skip MPIDIG_EPOCH_CHECK_SYNC if the target process is PROC_NULL.

Reply:
I see. I thought it was just about the code portion here. I have added a commit to skip all epoch checks for `PROC_NULL`. Does it look good?

Reply:
If `MPIDIG_RMA_OP_CHECK_SYNC` is always called after skip of `PROC_NULL` (e.g., in the third commit), you can skip the PROC_NULL check in the macro too. Can you please also change `MPIDIG_RMA_OP_CHECK_SYNC` in the third commit, and add comment saying that this macro assumes no `PROC_NULL` case calls it ?

Reply:
Done. I am re-running jenkins test to make sure the recent changes are good.